emperor/support_files/js/view-controller.js

```diff
@@ -247,35 +241,15 @@ define([
   EmperorViewController.prototype.constructor = EmperorViewControllerABC;
 
   /**
-   * Retrieves the name of the currently active decomposition view.
    *
-   * @return {String} A key corresponding to the active decomposition view.
-   */
-  EmperorViewController.prototype.getActiveDecompViewKey = function() {
-    return this.activeViewKey;
-  };
-
-  /**
-   * Changes the currently active decomposition view.
+   * Retrieve a view from the controller.
    *
-   * @param {String} k Key corresponding to active decomposition view.
-   * @throws {Error} The key must exist, otherwise an exception will be thrown.
-   */
-  EmperorViewController.prototype.setActiveDecompViewKey = function(k) {
-    if (this.decompViewDict[k] === undefined) {
-      throw new Error('This key does not exist, "' + k + '" in the ' +
-                      'the decompViewDict.');
-    }
-    this.activeViewKey = k;
-  };
-
-  /**
-   * Retrieves the currently active decomposition view.
+   * This class does not operate on single decomposition views, hence this
+   * method retrieves the first available view.
    *
-   * @return {DecompositionView} The currently active decomposition view.
    */
-  EmperorViewController.prototype.getActiveView = function() {
-    return this.decompViewDict[this.getActiveDecompViewKey()];
+  EmperorViewController.prototype.getView = function() {
+    return this.decompViewDict[Object.keys(this.decompViewDict)[0]];
```

Comment:
You are right, this was a temporary fix, but in this case, this method is for the intermediate EmperorViewController class, this class does not operate on a single view at any time, it does operations in bulk, however it is useful to get the zeroth view (as a reference). The controller that will be used for biplots for example, will be EmperorAttributeViewController, and in that one we can select which view we want.